박상우 PR(지각..)

coolfin opened the pull request:
늦어서 죄송합니다..ㅠ PR 목요일 전에 한 번 더 할게요!

coolfin committed:
INIT

coolfin committed:
FEAT: 메인 컨테이너
- add darkmode icon
- todolist container base design

- 다양한 곳에 사용되는 컴포넌트 사용을 위해 styled-components 추가

coolfin committed:
FEAT: redux
- todo의 CRUD를 redux로 set

coolfin committed:
DOCS

coolfin committed:
DOCS

coolfin committed:
FEAT:redux 연결
- map 활용 dispatch한 sample data 연결

coolfin committed:
FEAT: redux CRUD
- sample data CRUD 추가
- 추가되는 data에 따른 add button 동적 이동

coolfin committed:
FEAT: container status 분할
- combineReducers 사용, container 별 reducre 분할
- tailwind config 활용, 동적 배경 할당

bug : 각 container 별 data CRUD가 원활하게 이루어지지 않고 있음

coolfin committed:
Merge branch 'main' of https://github.com/coolfin/ssafe_todo

coolfin committed:
FEAT: redux CRUD

coolfin committed:
FEAT: list move
- 리스트 간 이동을 hover과 button을 활용하여 구현
- fix: 기능 구현 필요

kwakseongjae commented on src/App.jsx:
```diff
@@ -0,0 +1,67 @@
+  return (
+    <Fragment>
+      {/* Roadmap */}
+      <div className={classNames(
+        'w-full',
+        'py-12',
+        'px-8',
+
+        'text-[50px]',
+        'font-bold',
+        'text-theme-font',
+
+        'flex',
+        'justify-between',
+        'items-center',
+
+        'max-sm:flex-col',
+        'max-sm:justify-center',
+      )}>
```
tailwind 잘쓰네 !!

kwakseongjae commented on src/App.jsx:
```diff
@@ -0,0 +1,67 @@
+
+        {/* 다크모드 <-> 라이트모드 */}
+        <div onClick={handleDarkMode}>
+          <img src={isDarkMode ? "asset/icon/dark.png" : "asset/icon/light.png"} alt="darkmode" className={classNames(
```
지금은 이미지가 두개라서 괜찮긴한데, 나중엔 상단에 변수로 빼서 처리하면 더 좋을거같아 !

kwakseongjae commented on src/App.jsx:
```diff
@@ -0,0 +1,67 @@
+        'max-sm:gap-y-8'
+      )}>
+        {
+          ["To do", "In progress", "Done"].map((status, index) => (
```
상수 선언은 상단에 빼서 처리하거나 따로 파일을 만들어서 관리해보자 !

kwakseongjae commented on src/components/Container/index.jsx:
```diff
@@ -0,0 +1,114 @@
+
+            'gap-y-4',
+
+            
```
불필요한 줄바꿈은 최종본에선 다 빼줘 !

kwakseongjae commented on src/components/Container/index.jsx:
```diff
@@ -0,0 +1,114 @@
+                </div>
+
+                {/* 진행도 별 컨테이너 */}
+                {
+                    containerStatus === 'TODO' ? todos.map((todo, index) => (
+                        <Ticket key={index} status={containerStatus} todo={todo}/>
+                    )) : containerStatus === 'PROGRESS' ? progress.map((progress, index) => (
+                        <Ticket key={index} status={containerStatus} todo={progress}/>
+                    )) : containerStatus === 'DONE' ? done.map((done, index) => (
+                        <Ticket key={index} status={containerStatus} todo={done}/>
+                    )) : <span>Loading ... </span>
+                }
```
👍👍

kwakseongjae commented on src/store/reducer/index.js:
```diff
@@ -0,0 +1,13 @@
+import progressReducer from './progress';
+import doneReducer from './done';
+
+const rootReducer = combineReducers({
+    todos: todoReducer,
+    progress: progressReducer,
+    done: doneReducer
+});
+
+export default rootReducer;
```
👍👍

kwakseongjae reviewed (commented):
목요일까지 세부사항 손보면 좋은 결과물 나올거같네 😇
지금은 + 버튼을 계속 눌러서 BlackContainer밖으로 티켓이 생기면 색상이 다르게 보이는데, 
나중에 컨테이너 사이즈를 동적으로 변경시키거나 스크롤 기능 추가하면 해결해야할 문제는 아닌거같아 ! 

지금은 마우스 호버시에 버튼1, 버튼2가 보이는 카드가 렌더링되는데, 다음에는 수정이랑 X버튼도 한번 눌러보고 싶네 ㅎㅎ 
리덕스랑 테일윈드 두개 다 사용하려니 힘들텐데 목요일 까지 파이팅이야 !! 
방향은 맞게 가고 있는거같으니까 지금 구현하고자 하는 사항 구현에 집중해서 목요일에 보자 !!

coolfin committed:
FEAT: list move
- hover 시 디자인 수정
- ticket button component 추가

coolfin committed:
FEAT: ticket move

coolfin committed:
FEAT: ticket add modal
- ticket 추가 시 title, content 입력 modal 생성
- 컨테이너 색상 지정(예정)

fix: ticket move id 중복으로 인한 bug 해결

coolfin committed:
FEAT: add ticket
- 제목 및 내용을 입력하여 ticket을 추가하는 컴포넌트 구현

coolfin committed:
FEAT: edit ticket
- redux에서 정보를 받아와서 수정하는 기능 구현

coolfin committed:
FEAT: scroll
- index.css에 브라우저 별 스크롤 기능 추가
